src/Microsoft.AspNet.OData.Shared/Query/ODataQueryOptions.cs

```diff
@@ -964,14 +996,14 @@ private T ApplySelectExpand<T>(T entity, ODataQuerySettings querySettings)
                         SelectExpand.Context);
                 }
 
-                SelectExpandClause processedClause = SelectExpand.ProcessLevels();
+                SelectExpandClause processedClause = SelectExpand.ProcessedSelectExpandClause;
                 SelectExpandQueryOption newSelectExpand = new SelectExpandQueryOption(
                     SelectExpand.RawSelect,
                     SelectExpand.RawExpand,
                     SelectExpand.Context,
                     processedClause);
 
-                InternalRequest.Context.SelectExpandClause = processedClause;
+                InternalRequest.Context.ProcessedSelectExpandClause = processedClause;
```

Comment:
>ProcessedSelectExpandClause [](start = 40, length = 27)

Did we decide that we couldn't set QueryOptions here?

Reply:
Yes but there was a fallout because of that. As discussed, it is best to clean up all that with 8.x

---
In reply to: [292955379](https://github.com/OData/WebApi/pull/1724#discussion_r292955379) [](ancestors = 292955379)